trieste/models/model_interfaces.py

```diff
@@ -615,11 +614,19 @@ def randomize_model_hyperparameters(model: gpflow.models.GPModel) -> None:
     for param in model.trainable_parameters:
         if isinstance(param.bijector, tfp.bijectors.Sigmoid):
             sample = tf.random.uniform(
-                [1],
+                param.bijector.low.shape,
                 minval=param.bijector.low,
                 maxval=param.bijector.high,
                 dtype=param.bijector.low.dtype,
             )
             param.assign(sample)
         elif param.prior is not None:
             param.assign(param.prior.sample())
+
+
+def squeeze_hyperparameters(model: gpflow.models.GPModel, epsilon: float = 1e-8) -> None:
+    for param in model.trainable_parameters:
+        if isinstance(param.bijector, tfp.bijectors.Sigmoid):
+            squeezed_param = tf.math.minimum(param, param.bijector.high - epsilon)
+            squeezed_param = tf.math.maximum(squeezed_param, param.bijector.low + epsilon)
+            param.assign(squeezed_param)
```

Comment:
Perhaps add a little comment saying what this does/why it is needed? e.g. just use what you wrote in the PR description

Reply:
This might  also be unit-testable

Reply:
```suggestion
@random_seed
@pytest.mark.parametrize("dim", [1, 10])
def test_randomize_model_hyperparameters_samples_from_constraints_when_given_prior_and_constraint(
    dim: int,
) -> None:
    kernel = gpflow.kernels.RBF(variance=1.0, lengthscales=[0.2] * dim)
    upper = tf.cast([0.5] * dim, dtype=tf.float64)

    lower = upper / 100
    kernel.lengthscales = gpflow.Parameter(
        kernel.lengthscales, transform=tfp.bijectors.Sigmoid(low=lower, high=upper)
    )
    kernel.lengthscales.prior = tfp.distributions.Uniform(low=10.0, high=100.0)

    kernel.variance.prior = tfp.distributions.LogNormal(loc=np.float64(-2.0), scale=np.float64(1.0))

    randomize_model_hyperparameters(kernel)

    npt.assert_array_less(kernel.lengthscales, [0.5] * dim)
    npt.assert_raises(AssertionError, npt.assert_allclose, [0.2] * dim, kernel.lengthscales)
```

Reply:
(Yes, a docstring like for randomize_model_hyperparameters would be good.)

Reply:
thanks! done